### PR TITLE
Change suggest to client suggest

### LINKED ISSUE #356: Change "suggest" to "client suggest"

### PR DESCRIPTION
**Pull request by qwoprocks**
## Description

As per title.

Fixes #356

## Testing

Ran tests, all pass.

## Remarks

NIL.

### CODE REVIEW

File: src/main/java/seedu/address/model/client/ClientSuggestionType.java [hunk 1]
```diff
@@ -7,10 +7,10 @@
 import java.util.function.Predicate;
 
 /**
- * Represents a SuggestionType in the address book.
- * Guarantees: immutable; suggestion type is valid as declared in {@link #isValidSuggestionType(String)}
+ * Represents a ClientSuggestionType in the address book.
+ * Guarantees: immutable; suggestion type is valid as declared in {@link #isValidClientSuggestionType(String)}
```
Comment (tankangliang) [inline src/main/java/seedu/address/model/client/ClientSuggestionType.java]: `client suggestion type` here

File: docs/UserGuide.md [hunk 2]
```diff
@@ -450,27 +450,27 @@ Example:
 
     Deletes the country note at index 1 of the country notes list panel.
 
-### Getting suggestions on clients: `suggest`
+### Getting suggestions on clients: `client suggest`
 
 Obtains a list of clients based on the suggestion type(s) passed in. This command will also reset the left display panel to its default view.
```
Comment (tankangliang) [inline docs/UserGuide.md]: do we wanna change this to client suggestion type?

File: src/main/java/seedu/address/logic/commands/ClientSuggestCommand.java [hunk 3]
```diff
@@ -11,58 +11,58 @@
 
 import seedu.address.model.Model;
 import seedu.address.model.client.Client;
-import seedu.address.model.client.SuggestionType;
+import seedu.address.model.client.ClientSuggestionType;
 
 /**
  * Suggests a list of clients based on the specified suggestion types
```
Comment (tankangliang) [inline src/main/java/seedu/address/logic/commands/ClientSuggestCommand.java]: `client suggestion type` here?

File: src/main/java/seedu/address/logic/commands/ClientSuggestCommand.java [hunk 4]
```diff
@@ -11,58 +11,58 @@
...
+        requireNonNull(clientSuggestionTypeOrderedSet);
+        this.clientSuggestionTypeOrderedSet = clientSuggestionTypeOrderedSet;
+        clientSuggestTypeDescriptions = clientSuggestionTypeOrderedSet.stream()
+                .map(ClientSuggestionType::getDescription).collect(Collectors.joining("\n"));
+        this.clientSuggestionTypePredicate = getCombinedPredicate(clientSuggestionTypeOrderedSet);
+        this.clientSuggestionTypeCombinedComparator = getCombinedComparator(clientSuggestionTypeOrderedSet);
     }
 
     /**
      * Given an ordered set of suggestion types, returns a combined predicate that returns true only if the client
```
Comment (tankangliang) [inline src/main/java/seedu/address/logic/commands/ClientSuggestCommand.java]: `client suggestion type` here?

File: src/main/java/seedu/address/logic/commands/ClientSuggestCommand.java [hunk 5]
```diff
@@ -11,58 +11,58 @@
...
-        return suggestionTypeOrderedSet.stream()
-                .map(SuggestionType::getSuggestionPredicate)
+    private static Predicate<Client> getCombinedPredicate(Set<ClientSuggestionType> clientSuggestionTypeOrderedSet) {
+        return clientSuggestionTypeOrderedSet.stream()
+                .map(ClientSuggestionType::getClientSuggestionPredicate)
                 .reduce(client -> true, (p1, p2) -> client -> p1.test(client) && p2.test(client));
     }
 
     /**
      * Given an ordered set of suggestion types, returns a combined comparator that combines the suggestion types'
```
Comment (tankangliang) [inline src/main/java/seedu/address/logic/commands/ClientSuggestCommand.java]: `client suggestion type` here?

**Review by tankangliang — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/model/client/ClientSuggestionType.java (hunk 1 above)
Comment (qwoprocks) [reply src/main/java/seedu/address/model/client/ClientSuggestionType.java]: done

File: docs/UserGuide.md (hunk 2 above)
Comment (qwoprocks) [reply docs/UserGuide.md]: ok sure changed

File: src/main/java/seedu/address/logic/commands/ClientSuggestCommand.java (hunk 3 above)
Comment (qwoprocks) [reply src/main/java/seedu/address/logic/commands/ClientSuggestCommand.java]: done

File: src/main/java/seedu/address/logic/commands/ClientSuggestCommand.java (hunk 5 above)
Comment (qwoprocks) [reply src/main/java/seedu/address/logic/commands/ClientSuggestCommand.java]: done

**Review by tankangliang — APPROVED** [review]
LGTM

**Pull request merged**